stan/math/prim/scal/fun/lower_reg_inc_gamma.hpp

```diff
@@ -0,0 +1,194 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_LOWER_REG_INC_GAMMA_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_LOWER_REG_INC_GAMMA_HPP
+
+#include <boost/math/tools/promotion.hpp>
+#include <stan/math/prim/scal/fun/lgamma.hpp>
+#include <stan/math/prim/scal/fun/gamma_p.hpp>
+#include <stan/math/prim/scal/fun/inv.hpp>
+#include <stan/math/prim/scal/fun/digamma.hpp>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
+#include <stan/math/prim/scal/fun/grad_reg_inc_gamma.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
+#include <limits>
+
+namespace stan {
+  namespace math {
+
+    template <int IMPL = 0>
+    struct lower_reg_inc_gamma {
+      template <int ARG, typename T1, typename T2>
+      static
+      typename boost::math::tools::promote_args<T1, T2>::type
+      grad(T1 a, T2 z, double precision = 1e-10, int max_steps = 1e5) {
+        typedef typename boost::math::tools::promote_args<T1, T2>::type TP;
+        TP d = std::numeric_limits<TP>::quiet_NaN;
+        return d;
+      }
+    };
+
+
+    template <>
+    struct lower_reg_inc_gamma<1> {
+      template <int ARG, typename T1, typename T2>
+      static
+      typename boost::math::tools::promote_args<T1, T2>::type
+      grad(T1 a, T2 z, double precision = 1e-10,
+                                 int max_steps = 1e5) {
+        typedef typename boost::math::tools::promote_args<T1, T2>::type TP;
+
+        if (is_nan(a)) return std::numeric_limits<TP>::quiet_NaN();
+        if (is_nan(z)) return std::numeric_limits<TP>::quiet_NaN();
```

Comment:
There is a relatively clear policy:  any function other than a built-in C++ library function should throw with an informative error message as soon as it encounters an unrecoverable error.  We haven't done that consistently, but that's the policy.

We have toyed with changing the behavior of log, exp, etc., but the problem there is that if we do that, we potentially mess up other functions that use them and have to go back in and change the double implementations to make them consistent with the autodiff implementations.  So I originally decided it was worth following the NaN/inf policy of the library functions when implementing our autodiff to guarantee you always get the same value of a function using `double` and `var` instantiations.